### PR TITLE
Add checkstyle and reports

### PR DESCRIPTION
**Pull request by Cxristian**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+    // Task to define arguments and configurations to properties file
+    tasks.withType(Test) {
+
+    // enable TestNG support (default is JUnit)
+    useTestNG()
+        
+
+}
```
Comment (MelissaRoman06) [inline build.gradle]: Please add an empty line.

File: gradle.properties [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#type of browser chrome/firefox
+browser=chrome
+
+#Properties for GUI
+login=https://exampletest-trials653.orangehrmlive.com/
+username=Admin
+password=ywyDV2C5@C
```
Comment (MelissaRoman06) [inline gradle.properties]: You should not share your credentials.

File: src/main/java/core/selenium/webDrivers/FactoryBrowser.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+/**
+ * FactoryBrowser class.
+ *
+ * @author Cristian Lujan
+ * @version 0.0.1
+ */
+public final class FactoryBrowser {
+
+    private static WebDriverConfig webDriverConfig = WebDriverConfig.getInstance();
```
Comment (MelissaRoman06) [inline src/main/java/core/selenium/webDrivers/FactoryBrowser.java]: Initialization should not be here in the declaration of the attributes.

File: src/main/java/core/utils/Logs.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        return instance;
+    }
+
+    /**
+     * @return Logger LOG to do our loggers
+     */
+    public Logger getLog() {
+        return LOG;
+    }
+}
```
Comment (MelissaRoman06) [inline src/main/java/core/utils/Logs.java]: Please add an empty line.

File: src/main/java/core/utils/ReaderPropertiesFile.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+public class ReaderPropertiesFile {
+
+    private static final String URL_DRIVER_PROPERTIES = "driver.properties";
+    private static final String URL_GRADLE_PROPERTIES = "gradle.properties";
+
+    private Map<String, String> properties = new HashMap<>();
```
Comment (MelissaRoman06) [inline src/main/java/core/utils/ReaderPropertiesFile.java]: Initialization should not be here.

File: src/main/java/orangeHRM/entities/Context.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+    /**
+     * Returns the Courses of context.
+     *
+     * @return courses.
+     */
+    public Courses getCourses() {
+        return courses;
+    }
+}
```
Comment (MelissaRoman06) [inline src/main/java/orangeHRM/entities/Context.java]: Please add an empty line.

File: src/main/java/orangeHRM/entities/Courses.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+    /**
+     * Sets the subunit of Course.
+     *
+     * @param subunit for the account.
+     */
+    public void setSubunit(final String subunit) {
+        this.subunit = subunit;
+    }
+}
```
Comment (MelissaRoman06) [inline src/main/java/orangeHRM/entities/Courses.java]: Please add an empty line.

File: src/main/java/orangeHRM/entities/User.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+ * @author Cristian Lujan
+ * @version 0.0.1
+ */
+public class User {
+
+    private String userName;
+    private String email;
+    private String password;
+
+    public String getUserName() {
```
Comment (MelissaRoman06) [inline src/main/java/orangeHRM/entities/User.java]: Please add documentation in all methods.

File: src/main/java/orangeHRM/ui/pages/CoursePage.java [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+    /**
+     * Gets of name page.
+     *
+     * @return name of page.
+     */
+    public String getNamePage() {
+        return namePage.getText();
+    }
+}
```
Comment (MelissaRoman06) [inline src/main/java/orangeHRM/ui/pages/CoursePage.java]: Please add an empty line.

File: src/main/java/orangeHRM/ui/pages/DashboardPage.java [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+    /**
+     * Gets of title page.
+     *
+     * @return title of page.
+     */
+    public String getTitlePage() {
+        return titlePage.getText();
+    }
+
+}
```
Comment (MelissaRoman06) [inline src/main/java/orangeHRM/ui/pages/DashboardPage.java]: Please add an empty line.

File: src/test/java/steps/CourseStep.java [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+public class CourseStep {
+
+    PageTransporter pageTransporter = PageTransporter.getInstance();
+    private CoursePage coursePage;
+
+    @When("^I go to the Courses page$")
+    public void goToTheCoursesPage() {
+        coursePage = pageTransporter.navigateToCoursePage();
+    }
+}
```
Comment (MelissaRoman06) [inline src/test/java/steps/CourseStep.java:92]: Please add an empty line.

File: src/test/java/steps/CourseStep.java [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * steps.CourseStep class.
+ *
+ * @author Cristian Lujan
+ * @version 0.0.1
+ */
+public class CourseStep {
+
+    PageTransporter pageTransporter = PageTransporter.getInstance();
```
Comment (MelissaRoman06) [inline src/test/java/steps/CourseStep.java]: Initialization should no be here.

File: src/test/java/steps/DashboardStep.java [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+
+/**
+ * DashboardStep class.
+ *
+ * @author Cristian Lujan
+ * @version 0.0.1
+ */
+public class DashboardStep {
+
+    PageTransporter pageTransporter = PageTransporter.getInstance();
```
Comment (MelissaRoman06) [inline src/test/java/steps/DashboardStep.java]: Initialization should not be here.

File: src/test/java/steps/LoginSteps.java [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+
+/**
+ * LoginSteps class.
+ *
+ * @author Cristian Lujan
+ * @version 0.0.1
+ */
+public class LoginSteps {
+
+    PageTransporter pageTransporter = PageTransporter.getInstance();
```
Comment (MelissaRoman06) [inline src/test/java/steps/LoginSteps.java]: Initialization should not be here.

File: src/test/java/steps/LoginSteps.java [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+
+    /**
+     * Sign in step.
+     */
+    @When("^I am logged in orangeHRM site with (username) and (password) valid$")
+    public void logInOrangeHRMSiteWithUsernameAndPasswordValid(final String userName, final String password) {
+        loginPage = pageTransporter.navigateToLoginPage();
+        loginPage.login(userName, password);
+    }
+}
```
Comment (MelissaRoman06) [inline src/test/java/steps/LoginSteps.java]: Please add an empty line.

File: src/test/resources/features/Login.feature [hunk 16]
```diff
@@ -0,0 +1,7 @@
+
+Feature: Log In in the orangeHRM Application
+
+  Scenario: Login in the orangeHRM application
+    When I am logged in orangeHRM site with username and password valid
+    Then I am on the Dashboard page
+      And I see the name Dashboard on the dashboard page
```
Comment (MelissaRoman06) [inline src/test/resources/features/Login.feature]: Please add an empty line and align the indentation.

File: src/test/resources/features/Training.feature [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+        | Description | no                 |
+    Then I see a message that indicates the Course was created
+      And the Course title should be displayed in the course list in the Course page
+
+  Scenario: Delete a course that was created
+    Given I go to the Courses page
+    When I open Courses List page
+      And I select a Course exist in the Course List page
+      And I see a message to confirm
+    Then the course name don't should be displayed in the courses list of courses list page
```
Comment (MelissaRoman06) [inline src/test/resources/features/Training.feature]: Please add an empty line.

File: driver.properties [hunk 18]
```diff
@@ -0,0 +1,3 @@
+implicitWaitTime=30
```
Comment (MelissaRoman06) [inline driver.properties:1]: Please be consistent with the blank spaces after and before "="

File: src/main/java/core/selenium/WebDriverManager.java [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+/**
+ * WebDriverManager class.
+ *
+ * @author Cristian Lujan
+ * @version 0.0.1
+ */
+public class WebDriverManager {
+    private static WebDriver webDriver;
+    private WebDriverWait webDriverWait;
+    private WebDriverConfig webDriverConfig = WebDriverConfig.getInstance();
```
Comment (MelissaRoman06) [inline src/main/java/core/selenium/WebDriverManager.java]: Initialization shouldn't be here.

File: src/main/java/core/utils/DriverMethods.java [hunk 20]
```diff
@@ -0,0 +1,40 @@
...
+        webElement.click();
+        webElement.clear();
+        webElement.sendKeys(text);
+    }
+
+    /**
+     * Constructor private.
+     */
+    private DriverMethods() { }
+}
```
Comment (MelissaRoman06) [inline src/main/java/core/utils/DriverMethods.java:40]: Please add an empty line below.

File: src/main/java/orangeHRM/PageTransporter.java [hunk 21]
```diff
@@ -0,0 +1,105 @@
...
+    protected PageTransporter() {
+        initialize();
+    }
+
+    /**
+     * Gets instance of page transporter.
+     *
+     * @return a new instance if it was initialised.
+     */
+    public static PageTransporter getInstance() {
```
Comment (MelissaRoman06) [inline src/main/java/orangeHRM/PageTransporter.java:48]: PageTransporter should not be singleton.

File: src/test/java/steps/CourseStep.java [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+    Context context;
+    Courses courses;
+
+    public CourseStep(Context context) {
+        this.context = context;
+        courses = context.getCourses();
+
+    }
+
+    @When("^I go to the Courses page$")
```
Comment (MelissaRoman06) [inline src/test/java/steps/CourseStep.java:56]: Please add documentation to all methods.

File: src/test/java/steps/DashboardStep.java [hunk 23]
```diff
@@ -0,0 +1,49 @@
...
+ * @author Cristian Lujan
+ * @version 0.0.1
+ */
+public class DashboardStep {
+
+    PageTransporter pageTransporter;
+    DashboardPage dashboardPage;
+    WebDriver webDriver;
+
+    @Then("^I am on the Dashboard page$")
```
Comment (MelissaRoman06) [inline src/test/java/steps/DashboardStep.java:37]: Please add documentation in all methods.

File: build.gradle (hunk 1 above)
Comment (Cxristian) [reply build.gradle]: done

**Review by MelissaRoman06 — APPROVED** [review]

**Pull request merged**